cmd/minikube/cmd/config/disable.go

```diff
@@ -38,11 +40,22 @@ var addonsDisableCmd = &cobra.Command{
 		if addon == "heapster" {
 			exit.Message(reason.AddonUnsupported, "The heapster addon is depreciated. please try to disable metrics-server instead")
 		}
-		err := addons.SetAndSave(ClusterFlagValue(), addon, "false")
+		cc, err := config.Load(ClusterFlagValue())
 		if err != nil {
-			exit.Error(reason.InternalAddonDisable, "disable failed", err)
+			exit.Error(reason.InternalConfigView, "loading profile", err)
 		}
-		out.Step(style.AddonDisable, `"The '{{.minikube_addon}}' addon is disabled`, out.V{"minikube_addon": addon})
+		validAddon, ok := assets.Addons[addon]
+
+		if !ok {
+			exit.Message(reason.InternalAddonDisable, `"'{{.minikube_addon}}' is not a valid minikube addon`, out.V{"minikube_addon": addon})
```

Comment:
```suggestion
			exit.Message(reason.AddonUnsupported, `"'{{.minikube_addon}}' is not a valid minikube addon`, out.V{"minikube_addon": addon})
```
I think `AddonUnsupported` makes more sense here as it's not an internal error, but trying to use an invalid addon.